From eb97e2f2ed8c55caff8fac45528af7f8de19c1cb Mon Sep 17 00:00:00 2001
From: Frode Isaksen <fisaksex>
Date: Thu, 18 Oct 2012 12:07:31 +0200
Subject: [PATCH 40/78] wl12xx: Always log the packets that caused WiFi to
 wake up the platform

BZ: 62772

Finding why WiFi woke up the platform can be difficult involving using Wireshark or
Omnipeek network sniffers.
This patch will log the N packets (N=2) that is received after the WiFi chip
woke up the platform. With N=2, we will hopefully log the packet that
caused the wakeup, since typically we have an ARP packet and an IP packet.
The changes from patch 64576 is that the log is now printed as KERN_INFO as
to always have the log.

Change-Id: Icbef3d2cfd69cfa14323111f43ceddc43c27308d
Signed-off-by: Frode Isaksen <fisaksex>
Reviewed-on: http://android.intel.com:8080/70609
Reviewed-by: Isaksen, FrodeX <frodex.isaksen@intel.com>
Reviewed-by: De Cesco, Jonathan <jonathan.de.cesco@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/77997
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/main.c |    4 +---
 wl12xx-compat/drivers/net/wireless/wl12xx/rx.c   |    3 ++-
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
index 6d33b9f..c1d5aec 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
@@ -2341,9 +2341,7 @@ static int wl1271_op_resume(struct ieee80211_hw *hw)
 
 		/* don't talk to the HW if recovery is pending */
 		if (!pending_recovery) {
-			if (DEBUG_WAKE & wl12xx_debug_level) {
-				wl->log_wake_pkts = 2; /* Log 2 packets after wake, typically ARP + IP */
-			}
+			wl->log_wake_pkts = 2; /* Log 2 packets after wake, typically ARP + IP */
 			ret = wl12xx_irq_locked(wl);
 			if (ret)
 				wl12xx_queue_recovery_work(wl);
diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/rx.c b/wl12xx-compat/drivers/net/wireless/wl12xx/rx.c
index 0be586d..9f198e6 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/rx.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/rx.c
@@ -180,7 +180,8 @@ static int wl1271_rx_handle_data(struct wl1271 *wl, u8 *data, u32 length,
 	skb_trim(skb, skb->len - desc->pad_len);
 
 	if (!beacon && (wl->log_wake_pkts > 0)) {
-		wl1271_dump_ascii(DEBUG_WAKE, "wake: ", skb->data, min(skb->len, 128));
+		print_hex_dump(KERN_INFO, DRIVER_PREFIX "wake: ", DUMP_PREFIX_OFFSET,
+					   16, 1, skb->data, min_t(size_t, skb->len, 96), true);
 		--wl->log_wake_pkts;
 	}
 
-- 
1.7.9

